cylc/flow/task_events_mgr.py

```diff
@@ -679,60 +689,70 @@ def process_message(
         # Satisfy my output, if possible, and spawn children.
         # (first remove signal: failed/EXIT -> failed)
 
+        # Complete the corresponding task output, if there is one.
         msg0 = message.split('/')[0]
-        completed_trigger = itask.state.outputs.set_msg_trg_completion(
-            message=msg0, is_completed=True)
-        self.data_store_mgr.delta_task_output(itask, msg0)
-
-        # Check the `started` event has not been missed e.g. due to
-        # polling delay
-        if (message not in [self.EVENT_SUBMITTED, self.EVENT_SUBMIT_FAILED,
-                            self.EVENT_STARTED]
-                and not itask.state.outputs.is_completed(TASK_OUTPUT_STARTED)):
-            self.setup_event_handlers(
-                itask, self.EVENT_STARTED, f'job {self.EVENT_STARTED}')
-            self.spawn_func(itask, TASK_OUTPUT_STARTED)
+        if message.startswith(ABORT_MESSAGE_PREFIX):
+            msg0 = TASK_OUTPUT_FAILED
+
+        completed_output = None
+        if msg0 not in [TASK_OUTPUT_SUBMIT_FAILED, TASK_OUTPUT_FAILED]:
+            completed_output = itask.state.outputs.set_msg_trg_completion(
+                message=msg0, is_completed=True)
+            if completed_output:
+                self.data_store_mgr.delta_task_output(itask, msg0)
+
+        for implied in (
+            itask.state.outputs.get_incomplete_implied(msg0)
+        ):
+            # Set submitted and/or started first, if skipped.
+            # (whether by forced set, or missed message).
+            LOG.warning(
+                f"[{itask}] setting missed output: {implied}")
```

Comment:
I'm not sure that in all use cases this word makes sense, and just raises the question: "Why was it missed?"
```suggestion
                f"[{itask}] setting output: {implied}")
```

Reply:
Perhaps "implicit".

Reply:
The word refers to the live output message case. For example, if the scheduler receives "succeeded" without first receiving "started", then the started message was necessarily "missed" (as far as the scheduler is concerned) - network outage or whatever.  I'll consider rephrasing...

Reply:
Changed to "implied".